packages/delivery-service/src/index.ts

```diff
@@ -85,9 +85,7 @@ global.logger = winston.createLogger({
     if (!app.locals) {
         throw new Error('App has no locals');
     }
-    io.use(
-        socketAuth(app as express.Express & { locals: WithLocals['locals'] }),
-    );
+    io.use(socketAuth(app.locals.db, app.locals.web3Provider));
```

Comment:
in order to get rid of app.locals you can assign the vars attached to it directly to a const and pass it to the function
const web3Provider = await getWeb3Provider()
const db = await getDb()

io.use(socketAuth(db,web3Provider));